cpmodels/chess_piece_models_common.py

```diff
@@ -37,24 +38,28 @@ def build_model(base_model):
 
 def data_generators(
     preprocessing_function,
-    target_size,
-    batch_size,
-    train_path="../data/train/",
-    validation_path="../data/validation/",
+    target_size: tuple[int, int],
+    batch_size: int,
+    train_path: str = "../data/train/",
+    validation_path: str = "../data/validation/",
 ):
-    """
-    Returns the train and validation generators.
-
-    :param preprocessing_function: Corresponding preprocessing function
-        for the pretrained base model.
-        Example: from keras.applications.mobilenet_v2
-            import preprocess_input
-    :param target_size: The dimensions to which all images found will be
-        resized. Example: (224, 224)
+    """Return the train and validation generators.
+
+    :param preprocessing_function: Corresponding preprocessing function for the pretrained base model.
```

Comment:
Doesn't the black formatter take care of this? I would've hoped so

Reply:
It seems like they discarded this idea (https://stackoverflow.com/questions/70966298/python-black-code-formatter-doesnt-format-docstring-line-length), but they propose to use https://github.com/PyCQA/docformatter

Reply:
It's probably the easiest for the programmer to take care of this themself by customizing the color for docstrings and adding rulers in the editor (I'll add instructions on this to the contributing guidelines):
![image](https://github.com/davidmallasen/LiveChess2FEN/assets/51491203/de81c8b2-2505-445c-89d3-ea9bd83fead8)

One thing to note is that for each function's docstring, the function summary has to be one line long and each parameter
summary has to be one line long as well:
![image](https://github.com/davidmallasen/LiveChess2FEN/assets/51491203/bf970f84-586c-42ec-a88f-abd051d85951)

The function summary has to be one line long because of PEP-257's conventions. The parameter summary needs to be one line long because otherwise the parameter preview (at least in VS Code) would only display the first line:
![image](https://github.com/davidmallasen/LiveChess2FEN/assets/51491203/5efd130d-cc8b-42c3-b6a8-74490d6cb64e)

So in these cases, it requires a human's judgement to throw away some words or rephrase summaries.

It was not the easiest, but I finally got everything fixed in terms of the line length!

Reply:
Great job! Thanks for the rationale behind this too